src/DataSourceProvider.vue

```diff
@@ -565,6 +581,14 @@ export default {
               this.loadSelectedDataSource(this.selectedDataSource.id);
             }
 
+            break;
+          case 'click-save':
```

Comment:
The action in this `case` appears to be performing a validation. I would therefore use an event name that's related to validation. It means that even when a user doesn't click on save, they can trigger validation with a suitable name, and not have to trigger `click-save` just to run a validation.

I would at least expect the name to have the word `validate` or `validation`.

Reply:
> The action in this `case` appears to be performing a validation. I would therefore use an event name that's related to validation. It means that even when a user doesn't click on save, they can trigger validation with a suitable name, and not have to trigger `click-save` just to run a validation.
> 
> I would at least expect the name to have the word `validate` or `validation`.

makes sense, fixed